lib/puppet/provider/file/windows.rb

```diff
@@ -86,6 +88,32 @@ def validate
     end
   end
 
+  # munge the windows group permissions if the group is set to SYSTEM
+  #
+  # tl;dr: when SYSTEM user is the group and the resoure is not managing group then treat
+  # the resource as insync if System has FullControl access.
+  #
+  # On windows systems: files laid down when puppet is running as the SYSTEM user will have
+  # the SYSTEM user set as the "group" on them. In these situations we want puppet to ignore
+  # what the permissions should be as long as the system user has FullControl access.
```

Comment:
Nit: Probably best to document what return types are expected here. From looking at it, we expect an integer or nil

What does a nil integer mean?